models/training-tuning-scripts/root-cause-models/sequence_classifier.py

```diff
@@ -0,0 +1,254 @@
+        test_gdf["label"] = labels
+
+        test_dataset = Dataset(test_gdf)
+        test_dataloader = DataLoader(test_dataset, batchsize=batch_size)
+
+        eval_accuracy = 0
+        nb_eval_steps = 0
+        for df in test_dataloader.get_chunks():
+            b_input_ids, b_input_mask = self._bert_uncased_tokenize(df["text"], max_seq_len)
+            b_labels = torch.tensor(df["label"].to_array())
+            with torch.no_grad():
+                logits = self._model(b_input_ids, token_type_ids=None, attention_mask=b_input_mask)[0]
+
+            logits = logits.type(torch.DoubleTensor).to(self._device)
+            logits = cupy.fromDlpack(to_dlpack(logits))
+            label_ids = b_labels.type(torch.IntTensor).to(self._device)
+            label_ids = cupy.fromDlpack(to_dlpack(label_ids))
+            temp_eval_accuracy = self._flatten_accuracy(logits, label_ids)
+
+            eval_accuracy += temp_eval_accuracy
+            nb_eval_steps += 1
+
+        accuracy = eval_accuracy / nb_eval_steps
+
+        return float(accuracy)
+
+    def save_model(self, save_to_path="."):
+        """
+        Save trained model
+
+        :param save_to_path: directory path to save model, default is current directory
+        :type save_to_path: str
+
+        Examples
+        --------
+        >>> from cuml.preprocessing.model_selection import train_test_split
+        >>> emails_train, emails_test, labels_train, labels_test = train_test_split(train_emails_df,
+                                                                                    'label',
+                                                                                    train_size=0.8)
+        >>> sc.train_model(emails_train, labels_train)
+        >>> sc.save_model()
+        """
+
+        self._model.module.save_pretrained(save_to_path)
+
+    def save_checkpoint(self, file_path):
+        """
+        Save model checkpoint
+
+        :param file_path: file path to save checkpoint
+        :type file_path: str
+
+        Examples
+        --------
+        >>> sc.init_model("bert-base-uncased")  # huggingface pre-trained model
+        >>> sc.train_model(train_data, train_labels)
+        >>> sc.save_checkpoint(PATH)
+        """
+
+        checkpoint = {"state_dict": self._model.module.state_dict()}
+        torch.save(checkpoint, file_path)
+
+    def load_checkpoint(self, file_path):
+        """
+        Load model checkpoint
+
+        :param file_path: file path to load checkpoint
+        :type file_path: str
+
+        Examples
+        --------
+        >>> sc.init_model("bert-base-uncased")  # huggingface pre-trained model
+        >>> sc.load_checkpoint(PATH)
+        """
+
+        model_dict = torch.load(file_path)
+        self._model.module.load_state_dict(model_dict["state_dict"])
+
+    def _get_hash_table_path(self):
+        hash_table_path = "%s/resources/bert-base-uncased-hash.txt" % os.path.dirname(os.path.realpath(__file__))
```

Comment:
this path is invalid within morpheus repo

Reply:
I've created that file and folder which we also had for phishing detection. so they follow the same convention now